app/views/antminer.py

```diff
@@ -18,11 +18,11 @@
 from datetime import timedelta
 import time
 
-# Update from one unit to the next if the value is greater than 1000.
-# e.g. update_unit_and_value(1000, "GH/s") => (1, "TH/s")
+# Update from one unit to the next if the value is greater than 1024.
+# e.g. update_unit_and_value(1024, "GH/s") => (1, "TH/s")
 def update_unit_and_value(value, unit):
-    if value > 1000:
-        value = value / 1000
+    while value > 1024:
+        value = value / 1024
```

Comment:
You forgot to add the dot at the end like `value = value / 1024.`, thus `value` is still an interer and not a float.

> You sure about the 1024? I thought that would apply just for when we deal with bytes.

I got your comment about the 1000 vs 1024 conversion and maybe you are right. Why are you in such a rush ? We can always discuss things.
But since you changed it to 1024 I guess you agree with me

Reply:
In the end i thought hashrate is probably related to clockrate which is driven by powers of 1024